cryptographic_estimators/MREstimator/MRAlgorithms/support_minors.py

```diff
@@ -0,0 +1,261 @@
+        self.set_parameter_ranges('nprime', 1, n)
+
+    @optimal_parameter
+    def a(self):
+        """
+        Return the optimal `a`, i.e. no. of vectors to guess in the kernel of the low-rank matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.a()
+            1
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS)
+
+    @optimal_parameter
+    def lv(self):
+        """
+        Return the optimal `lv`, i.e. no. of entries to guess in the solution
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.lv()
+            4
+        """
+        return self._get_optimal_parameter(MR_NUMBER_OF_COEFFICIENTS_TO_GUESS)
+
+    @optimal_parameter
+    def b(self):
+        """
+        Return the optimal `b`, i.e. the degree of the linear variables in the Macaulay matrix
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.b()
+            1
+        """
+        return self._get_optimal_parameter(MR_LINEAR_VARIABLES_DEGREE)
+
+    @optimal_parameter
+    def nprime(self):
+        """
+        Return the optimal `nprime`, i.e. the number of columns to be selected
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            4
+        """
+        return self._get_optimal_parameter(MR_REDUCED_NUMBER_OF_COLUMNS)
+
+    @optimal_parameter
+    def variant(self):
+        """
+        Return the optimal `variant`
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MREstimator.MRAlgorithms.support_minors import SupportMinors
+            sage: from cryptographic_estimators.MREstimator.mr_problem import MRProblem
+            sage: SM = SupportMinors(MRProblem(q=7, m=9, n=10, k=15, r=4))
+            sage: SM.nprime()
+            'strassen'
+        """
+        return self._get_optimal_parameter(MR_VARIANT)
+
+
+    def _expected_dimension_of_support_minors_equations(self,q, m, n, K, r, b):
```

Comment:
I can imagine that this functions is quite slow. Would it be better to translate all `* binomial(x, y)` to `+ log2(...)`?

Reply:
Notice that `* binomial(x, y)` are inside a sum. That is one has `sum([* binomial(x, y)])`. How to apply `logs` in there?

Reply:
I mean if it is fast enough, all good. Otherwise the question is how important this level of accuracy is and if we could provide a faster variant via some approximation.